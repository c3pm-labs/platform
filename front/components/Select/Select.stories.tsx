import React from 'react';
import { Formik, Form } from 'formik';
import { Meta } from '@storybook/react';
import { actions } from '@storybook/addon-actions';

import Select from '.';

export default {
  title: 'Select',
} as Meta;

const onSubmit = actions('onSubmit');

const options = [
  { value: 'gabriel', label: 'Gabriel' },
  { value: 'clement', label: 'ClÃ©ment' },
  { value: 'itagiba', label: 'Itagiba' },
];

export const Basic = (): JSX.Element => (
  <Formik
    initialValues={{ user: options[0].value }}
    {...onSubmit}
  >
    <Form>
      <Select name="user" options={options} />
    </Form>
  </Formik>
);
