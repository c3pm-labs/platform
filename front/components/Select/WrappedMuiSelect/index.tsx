import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';
import FormControl from '@material-ui/core/FormControl';
import MenuItem from '@material-ui/core/MenuItem';
import MuiSelect from '@material-ui/core/Select';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';
import { FieldInputProps } from 'formik';

export interface WrappedMuiSelectProps {
  selectOptions: Array<{ label: string; value: string }>;
  // eslint-disable-next-line @typescript-eslint/no-explicit-any
  field: FieldInputProps<any>;
}

const useStyles = makeStyles((theme: Theme) => createStyles({
  default: {
    width: 93,
    height: 30,
    fontSize: 14,
    textAlign: 'center',
    color: theme.palette.text.primary,
  },
  icon: {
    fill: theme.palette.text.primary,
    fontSize: 'large',
    height: '80%',
    marginRight: 5,
  },
  root: {
    paddingLeft: 10,
    paddingRight: 10,
  },
  select: {
    backgroundColor: 'rgba(0, 0, 0, 0)',
  },
  menuItem: {
    width: 93,
    height: 30,
    fontSize: 14,
    background: theme.palette.grey[100],
    color: theme.palette.text.primary,
    textOverflow: 'ellipsis',
    textAlign: 'center',
  },
  container: {
    width: '100%',
    overflow: 'hidden',
    textOverflow: 'ellipsis',
    textAlign: 'center',
  },
  selectMenuItem: {
    fontWeight: 'bold',
  },
  listMenu: {
    padding: 0,
    margin: 0,
  },
}));

function WrappedSelect({ selectOptions, field }: WrappedMuiSelectProps): JSX.Element {
  const classes = useStyles();

  return (
    <FormControl className={classes.default}>
      <MuiSelect
        disableUnderline
        IconComponent={ExpandMoreIcon}
        autoWidth={false}
        MenuProps={
            {
              classes: { list: classes.listMenu },
              elevation: 0,
              variant: 'menu',
              anchorOrigin: { vertical: 'bottom', horizontal: 'left' },
              getContentAnchorEl: null,
              transformOrigin: { vertical: 'top', horizontal: 'left' },
            }
          }
        className={classes.default}
        inputProps={{
          classes: {
            icon: classes.icon,
            select: classes.select,
            root: classes.root,
          },
        }}
        {...field}
      >
        {
          selectOptions.map((selectItemOptions) => (
            <MenuItem
              key={selectItemOptions.value}
              value={selectItemOptions.value}
              className={classes.menuItem}
              classes={{
                selected: classes.selectMenuItem,
              }}
            >
              <div className={classes.container}>
                {selectItemOptions.label}
              </div>
            </MenuItem>
          ))
        }
      </MuiSelect>
    </FormControl>
  );
}

export default WrappedSelect;
