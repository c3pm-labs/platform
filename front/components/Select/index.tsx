import { Field } from 'formik';

import WrappedSelect from './WrappedMuiSelect';

export interface SelectProps {
  options: Array<{ label: string; value: string }>;
  name: string;
}

function Select({
  options, name,
}: SelectProps): JSX.Element {
  return (
    <Field
      name={name}
      component={WrappedSelect}
      selectOptions={options}
    />
  );
}

export default Select;
