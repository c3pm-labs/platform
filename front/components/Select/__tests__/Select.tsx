import { render } from '@testing-library/react';
import React from 'react';
import { Formik, Form } from 'formik';

import Select from '../index';

test('Select', () => {
  const options = [
    { value: 'gabriel', label: 'Gabriel' },
    { value: 'clement', label: 'Cl√©ment' },
    { value: 'itagiba', label: 'Itagiba' },
  ];

  const { container } = render(
    <Formik
      initialValues={{ user: options[0].value }}
      // eslint-disable-next-line no-console
      onSubmit={() => console.log('select')}
    >
      <Form>
        <Select name="user" options={options} />
      </Form>
    </Formik>,
  );

  expect(container).toMatchSnapshot();
});
