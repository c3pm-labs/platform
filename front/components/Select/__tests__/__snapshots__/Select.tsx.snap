// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select 1`] = `
<div>
  <form
    action="#"
  >
    <div
      class="MuiFormControl-root makeStyles-default-1"
    >
      <div
        class="MuiInputBase-root MuiInput-root makeStyles-default-1 MuiInputBase-formControl MuiInput-formControl"
      >
        <div
          aria-haspopup="listbox"
          aria-labelledby="mui-component-select-user"
          class="MuiSelect-root makeStyles-root-3 MuiSelect-select makeStyles-select-4 MuiSelect-selectMenu MuiInputBase-input MuiInput-input"
          id="mui-component-select-user"
          role="button"
          tabindex="0"
        >
          <div
            class="makeStyles-container-6"
          >
            Gabriel
          </div>
        </div>
        <input
          aria-hidden="true"
          class="MuiSelect-nativeInput"
          name="user"
          tabindex="-1"
          value="gabriel"
        />
        <svg
          aria-hidden="true"
          class="MuiSvgIcon-root MuiSelect-icon makeStyles-icon-2"
          focusable="false"
          viewBox="0 0 24 24"
        >
          <path
            d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
          />
        </svg>
      </div>
    </div>
  </form>
</div>
`;
